src/qibolab/instruments/zhinst.py

```diff
@@ -1169,15 +1169,16 @@ def rearrange_sweepers(sweepers: List[Sweeper]) -> Tuple[np.ndarray, List[Sweepe
         rearranging_axes = np.zeros(2, dtype=int)
         if len(sweepers) == 2:
             if sweepers[1].parameter is Parameter.frequency:
-                if (sweepers[0].parameter is Parameter.bias) or (
-                    not sweepers[0].parameter is Parameter.amplitude
-                    and sweepers[0].pulses[0].type is not PulseType.READOUT
-                ):
-                    rearranging_axes[:] = [sweepers.index(sweepers[1]), 0]
-                    sweeper_changed = sweepers[1]
-                    sweepers.remove(sweeper_changed)
-                    sweepers.insert(0, sweeper_changed)
-                    log.warning("Sweepers were reordered")
+                if not sweepers[0].pulses is None:
+                    if (sweepers[0].parameter is Parameter.bias) or (
+                        not sweepers[0].parameter is Parameter.amplitude
+                        and sweepers[0].pulses[0].type is not PulseType.READOUT
+                    ):
+                        rearranging_axes[:] = [sweepers.index(sweepers[1]), 0]
+                        sweeper_changed = sweepers[1]
+                        sweepers.remove(sweeper_changed)
+                        sweepers.insert(0, sweeper_changed)
```

Comment:
```suggestion
                        sweepers = sweepers[::-1]
```
Since it is assumed that the length of sweepers is no more that 2, then this is a simpler way achieve the same result